app/tests/integration/nimbus/test_e2e_create_experiment.py

```diff
@@ -126,3 +129,95 @@ def test_create_new_experiment_remote_settings(selenium, base_url):
+    experiment.hypothesis = "smart stuff here"
+    experiment.application = "DESKTOP"
+
+    # Fill Overview Page
+    overview = experiment.save_and_continue()
+    overview.public_description = "description stuff"
+    overview.select_risk_brand_false()
+    overview.select_risk_revenue_false()
+    overview.select_risk_partner_false()
+
+    # Fill Branches page
+    branches = overview.save_and_continue()
+    branches.remove_branch()
+    branches.reference_branch_name = "name 1"
+    branches.reference_branch_description = "a nice experiment"
+    branches.feature_config = "No Feature Firefox Desktop"
+
+    # Fill Metrics page
+    metrics = branches.save_and_continue()
+
+    # Fill Audience page
+    audience = metrics.save_and_continue()
+    audience.channel = "Nightly"
+    audience.min_version = 80
+    audience.targeting = "US_ONLY"
+    audience.percentage = 50.0
+    audience.expected_clients = 50
+    audience.save_btn()
+    review = audience.save_and_continue()
+
+    # Review and approve
+    selenium.find_element_by_css_selector("#PageRequestReview")
+    review.launch_without_preview.click()
+    review.request_review.click_launch_checkboxes()
+    review.request_review.request_launch_button.click()
+    review.approve()
+    selenium.get("http://kinto:8888/v1/admin")
+    kinto_login = Login(selenium, base_url).wait_for_page_to_load()
+    kinto_login.kinto_auth.click()
+    kinto_dashbard = kinto_login.login()
+    bucket = kinto_dashbard.buckets[-1]
+    for item in bucket.bucket_category:
+        if "nimbus-desktop-experiments" in item.text:
+            item.click()
+            break
+    record = kinto_dashbard.record
+    record.action("reject")
+    kinto_dashbard = Dashboard(selenium, base_url)
+    modal = kinto_dashbard.reject_modal
+    modal.decline_changes()
+
+    # Load home page and wait for experiment to show in the Drafts tab
+    selenium.get(base_url)
+    for attempt in range(45):
+        try:
+            home = HomePage(selenium, base_url).wait_for_page_to_load()
+            home.tabs[-1].click()
+            new_experiments = len(home.tables[0].experiments)
+            assert new_experiments != current_experiments
+        except AssertionError:
+            time.sleep(2)
+            selenium.refresh()
+        else:
+            home.tabs[-1].click()
+            draft_experiments = home.tables[0]
+            for item in draft_experiments.experiments:
+                if experiment_name in item.text:
+                    item.click()
+                    break
+        finally:
+            break
+    experiment_url = experiment_name.replace(" ", "-")
+    print(experiment_url)
+    selenium.get(f"{base_url}/{experiment_url}/request-review")
+    for attempt in range(30):
+        try:
+            summary_page = ReviewPage(selenium, base_url).wait_for_page_to_load()
+            assert summary_page.rejected_text, "Rejected text box did not load"
```

Comment:
Sorry, I'm not good w/ Python...

If we have a test, and the `try` succeeds and our assert passes and no exception is thrown on the first attempt... is that it? Or will it keep doing the `for..in` loop another 29 times? Do we need to break out of the loop somehow on success? Or am I missing something?

Reply:
Another good catch, I should exit the loop using an else clause.